### PR TITLE
[SPARK-16236] [SQL] [FOLLOWUP] Add Path Option back to Load API in DataFrameReader

### PR DESCRIPTION
**Pull request by gatorsmile**
#### What changes were proposed in this pull request?

In Python API, we have the same issue. Thanks for identifying this issue, @zsxwing ! Below is an example:

``` Python
spark.read.format('json').load('python/test_support/sql/people.json')
```
#### How was this patch tested?

Existing test cases cover the changes by this PR

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 1]
```diff
@@ -244,7 +244,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   def json(path: String): DataFrame = {
     // This method ensures that calls that explicit need single argument works, see SPARK-16009
-    json(Seq(path): _*)
+    option("path", path).json(Seq.empty: _*)
```
Comment (zsxwing) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: @gatorsmile setting `path` or `paths` are both fine for json, parquet, and other file formats. See https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L312

You don't need to change this file.

Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: @zsxwing I see your point. The issue is from the others who build the data sources using the data source APIs. See the discussion in https://github.com/apache/spark/pull/13727#discussion_r68624316

Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: nvm. I found the issue. Json, parquet and the other file formats will not affect the data source API developers. : ) Let me revert the changes.

**Comment by zsxwing** [comment]
LGTM. Merging to master and 2.0. Thanks!
